lib/db.js

```diff
@@ -172,24 +187,46 @@ class LimitDBRedis extends EventEmitter {
       });
     }
 
-    this.redis.take(`${params.type}:${params.key}`,
-      bucketKeyConfig.per_interval / bucketKeyConfig.interval || 0,
+    if (cacheEnabled) {
+      const cached = this.cache.get(key);
+      if (cached) {
+        return callback(null, cached);
+      }
+    }
+
+    this.redis.take(key,
+      bucketKeyConfig.ms_per_interval || 0,
       bucketKeyConfig.size,
       count,
-      Math.floor(bucketKeyConfig.ttl || this.globalTTL),
+      Math.ceil(bucketKeyConfig.ttl || this.globalTTL),
+      bucketKeyConfig.drip_interval || 0,
       (err, results) => {
         if (err) {
           return callback(err);
         }
 
         const remaining = parseInt(results[0], 10);
-        return callback(null, {
-          conformant: parseInt(results[1], 10) ? true : false,
+        const conformant = parseInt(results[1], 10) ? true : false;
+        const currentMS = parseInt(results[2], 10);
+        const reset = parseInt(results[3], 10);
+        const res = {
+          conformant,
           remaining,
-          reset: this.calculateReset(bucketKeyConfig, remaining, parseInt(results[2], 10)),
+          reset: Math.ceil(reset / 1000),
           limit: bucketKeyConfig.size,
           delayed: false
-        });
+        };
```

Comment:
Does it make sense to add property if this was answered from Redis or the cache? If we want to instrument that to check how this cache is working

Reply:
If we are ok with extending the response, I see no issue.